src/main/java/slimeknights/tconstruct/tables/client/inventory/module/SideButtonsWidget.java

```diff
@@ -0,0 +1,120 @@
+package slimeknights.tconstruct.tables.client.inventory.module;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import net.minecraft.client.gui.components.Button;
+import net.minecraft.client.gui.components.Widget;
+import net.minecraft.client.gui.components.events.GuiEventListener;
+import net.minecraft.client.renderer.Rect2i;
+import org.apache.commons.compress.utils.Lists;
+import slimeknights.mantle.client.screen.MultiModuleScreen;
+
+import java.util.List;
+
+public class SideButtonsWidget<T extends Button> implements Widget, GuiEventListener {
+
+  protected final MultiModuleScreen<?> parent;
+
+  // left or right of the parent
+  protected final boolean right;
+
+  public int leftPos;
+  public int topPos;
+  public int imageWidth;
+  public int imageHeight;
+  public int xOffset;
+  public int yOffset;
+
+  private final int columns;
+  protected final List<T> buttons = Lists.newArrayList();
+  private Button clickedButton;
+
+  public int spacing = 4;
```

Comment:
While we are refactoring this, I do not like random public non-static variables. This should either be private or static final.